### PR TITLE
Calendar fix

### PR DESCRIPTION
**Pull request by alexr-bq**
### Summary

Replaces method of setting configuration value for how to handle old date values.

### Description

The old method of setting configuration failed under certain circumstances. This fixes that issue. This also changes the cleanup logic to clean up the parent unique-generated directory.

### Additional Reviewers

@jonathanl-bq

### CODE REVIEW

File: connector/src/main/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemWritePipe.scala [hunk 1]
```diff
@@ -64,6 +68,19 @@ class VerticaDistributedFilesystemWritePipe(val config: DistributedFilesystemWri
...
   }
 
+  /**
+   * Set spark conf to handle old dates if unset
+   * This deals with SPARK-31404 -- issue with legacy calendar format
+   */
+  private def setSparkkCalendarConf(): Unit = {
+    SparkSession.getActiveSession match {
+      case Some(session) =>
+        //session.conf.set(SQLConf.LEGACY_PARQUET_REBASE_MODE_IN_WRITE.key , "CORRECTED")
```
Comment (alexey-temnikov) [inline connector/src/main/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemWritePipe.scala:78]: nit: should we remove this comment?

Comment (alexr-bq) [reply connector/src/main/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemWritePipe.scala:78]: Yep

File: connector/src/main/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemWritePipe.scala [hunk 2]
```diff
@@ -64,6 +68,19 @@ class VerticaDistributedFilesystemWritePipe(val config: DistributedFilesystemWri
     }
   }
 
+  /**
+   * Set spark conf to handle old dates if unset
+   * This deals with SPARK-31404 -- issue with legacy calendar format
+   */
+  private def setSparkkCalendarConf(): Unit = {
```
[the remark this answers is not in the document]
Comment (alexr-bq) [reply connector/src/main/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemWritePipe.scala:75]: Yep

File: connector/src/main/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemWritePipe.scala [hunk 3]
```diff
@@ -102,6 +121,7 @@ class VerticaDistributedFilesystemWritePipe(val config: DistributedFilesystemWri
 
       // Create job status table / entry
       _ <- tableUtils.createAndInitJobStatusTable(config.tablename, config.jdbcConfig.username, config.sessionId, if(config.isOverwrite) "OVERWRITE" else "APPEND")
+
```
Comment (jonathanl-bq) [inline connector/src/main/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemWritePipe.scala:124]: remove

**Review by jonathanl-bq — APPROVED** [review]

**Pull request merged**